src/LegacyPaymentGateways/Adapters/LegacyPaymentGatewayAdapter.php

```diff
@@ -72,6 +73,10 @@ public function handleBeforeGateway(array $legacyDonationData, PaymentGatewayInt
                 'donationFormId' => $formData->formId
             ]);
 
+            $donation->subscriptionId = $subscription->id;
+            $donation->save();
```

Comment:
@ravinderk the subscriptionId is the `subscription_id` meta key.  from my research this is only used for renewals. The initial subscription's donation needs this to be set as 0

Reply:
fixed: https://github.com/impress-org/givewp/compare/345a253d392c...388b14f8ed17